internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriter.java

```diff
@@ -485,19 +429,32 @@ public CompletableFuture<VeniceResourceCloseResult> closeAsync(
                 logger.warn("Swallowed an exception while trying to end all segments for topic: {}", topicName, e);
               }
             }
-            producerAdapter.close(topicName, closeTimeOutInMs, gracefulClose);
+
+            long timeLeftToClose = 0; // N.B.: 0 indicates an ungraceful close
+            if (gracefulClose) {
+              /** N.B.: We measure the time left to try to adhere strictly to the configured {@link closeTimeOutInMs} */
+              long timeSinceStartOfClose = System.currentTimeMillis() - timeAtStartOfClose;
+              timeLeftToClose = Math.max(0, closeTimeOutInMs - timeSinceStartOfClose);
+            }
+
+            /**
+             * N.B.: We call the overload with a topic name param here for the sake of shared producer...
+             * TODO: Decide whether to keep shared producer or not...
```

Comment:
We've [deleted shared producer code](https://github.com/linkedin/venice/pull/923) sometime back. I think we can remove topic name from the close call

Reply:
Sure, I can do that, then.